test/ready/shared.dart

```diff
@@ -28,8 +28,14 @@ void sharedTests() {
     // Subscribe to the events.
     watcher.events.listen((event) {});
 
-    // Should eventually be ready.
-    await watcher.ready;
+    // Allow watcher to become ready
+    await pumpEventQueue();
+
+    // Ensure ready completes immediately
+    expect(
+        watcher.ready.timeout(new Duration(milliseconds: 0),
+            onTimeout: () => throw 'Does not complete immedately'),
```

Comment:
This is a race condition. A watcher becoming ready involves OS interaction—there's no guarantee that a `pumpEventQueue()` call will last long enough for it to happen.

This test case should be essentially the same as it was in the `scheduled_test` world, with a `ready` variable and everything.